### ISSUE TITLE
Seems there is memory leak

### ISSUE THREAD

**Issue by williamyang7577** [ghettovoice/gosip#9]
hi Ghettovoice，
Sorry， Seems I made  a mistake to close the previous issue. here. I had to resubmit a new one.

Below is my test code, please check.
[test.zip](https://github.com/ghettovoice/gosip/files/4641810/test.zip)
![image](https://user-images.githubusercontent.com/57163333/82170941-788c6680-98f8-11ea-8ff0-a18bc71f3687.png)

B.R
William

**Comment by ghettovoice** [comment]
Re-open of #8

**Comment by ghettovoice** [comment]
@williamyang7577 thanks for sample. I'll try with it soon.

**Comment by williamyang7577** [comment]
Hi Ghettovice，
May I know if having some solutions for this memory leak? 

B.R
William

**Comment by ghettovoice** [comment]
@williamyang7577 
yes, I have made several tests, but with my own simplified main script (because your is linking to your local packages). And I didn't saw any abnormal memory increasing, neither in the pprof nor in the system `htop`.
As I see from your screenshot you are using `-alloc_space` type, but this type show total amount of memory allocated (regardless of released by gc). If you view `-inuse_space` than you can see that nothing settle as a dead weight in memory. 
For example if you run with `GOGC=off` then you will see constantly increasing memory inuse.

This is my main script that I used with latest `gosip` version built from master.
[gosip.go.zip](https://github.com/ghettovoice/gosip/files/4763593/gosip.go.zip)
